crypto/aes_gcm.go

```diff
@@ -0,0 +1,167 @@
+package crypto
+
+import (
+	"crypto/aes"
+	"crypto/cipher"
+	"crypto/rand"
+	"crypto/sha512"
+	"encoding/json"
+	"fmt"
+	"hash"
+
+	"github.com/bndw/pick/errors"
+	"golang.org/x/crypto/pbkdf2"
+)
+
+type AESGCMClient struct {
```

Comment:
I can't think of a good reason to export these three structs. 

Further, due to the risk of misconfiguring, I'd expect any call site outside of this package to use a `New` method to instantiate one of these structs, if that should become necessary.